res/controllers/midi-components-0.0.js

```diff
@@ -348,13 +351,23 @@
         },
         connect: function() {
             Button.prototype.connect.call(this); // call parent connect
-            if (undefined !== this.group && this.colorKey !== undefined) {
+            if (this.group === undefined) {
+                return;
+            }
+            if (this.colorKey !== undefined) {
                 this.connections[1] = engine.makeConnection(this.group, this.colorKey, function(color) {
-                    if (engine.getValue(this.group, this.outKey)) {
+                    if (this.outGetValue()) {
                         this.outputColor(color);
                     }
                 });
             }
+            // Users can specify their own highlight(active) function
+            // so the mapping can respond when hotcues are currently active.
+            // The User should not call `trigger` in their highlight function
+            // to avoid crashing from endless recursion.
+            if (typeof this.highlight === "function") {
+                this.connections[2] = engine.makeConnection(this.group, "hotcue_" + this.number + "_activate", this.highlight);
```

Comment:
Did you also consider the shift feature? 

The "_activate" control is the control that is normally bound to the pressed state of the cue button. Not original designed for feedback. 
We already have also the "enabled" alias "status" CO, that has: 
https://github.com/mixxxdj/mixxx/blob/d993ec5bab09533dd124dbbdd039ddb131fafee0/src/engine/controls/cuecontrol.h#L61-L72

Calling the callback "highlight" puts implementation details of the callback into the name. I would prefer to call it according to the event when it is called. like "onActiveChanged" or such. 
In this case "onStatusChanged"?

Reply:
This is not really what its doing though. It's only highlighting the cue while its being pressed. So yeah its not designed for feedback, but it does tell us when the hotcue button is being pressed and when its being released. The `enabled` CO does not carry the information intended.  

> Did you also consider the shift feature?

Good point, I didn't, I'll have to test how that behaves, but I don't think it makes sense to have this in the shift state because then, there is no hotcue thats actually being activated.

Reply:
Ok, so the issue is that the name of the callback does not reveal which state is highlighted. It can be called "pressed" 
But that also reveals Implementation details, because you can also use the cue activate control without releasing. 

It feels wrong to to me to introduce here a convenient callback without having one for the "status" Co used in skins. We should encourage the mapping authors to use that instead. For this, IMHO corner case, the mapping authors can connect directly to the activate control. 

Can you please outline in a table for hot and cure loops why we need to highlight the pressed state? Maybe we can implement it by extending the status Co even more and that way also skins can benefit from it. 

@ronso0 what is you opinion?

Reply:
We don't _need_ to highlight the pressed state. It's simply a cosmetic feature requested in #4027 that I thought is better solved upstream instead of the mapping itself.
I'll look into what it takes to use `hotcue_X_status` instead. It could simplify the hotcuebutton component in general.

Reply:
So because the user can overwrite `outValueScale`, we can't really change the `outKey` nor easily simplify the button. ComponentsJS needs a refactoring (with breaking changes) for 2.4 anyways, so I'll postpone this change until then.

Reply:
fyi, I wrote down a bunch of thoughts in the wiki if you're interested. I don't plan on acting on them though in the near future. Getting 2.4 stable has priority. https://github.com/mixxxdj/mixxx/wiki/Component-JS-ES7-Brainstorming